streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -315,12 +315,14 @@ public void initializeTopology() {
     public void resume() {
         log.debug("Resuming");
         if (eosEnabled) {
-            if (producer != null) {
-                throw new IllegalStateException("Task producer should be null.");
+            if (!eosBetaEnabled) {
+                if (producer != null) {
+                    throw new IllegalStateException("Task producer should be null.");
+                }
+                producer = producerSupplier.get();
+                initializeTransactions();
```

Comment:
qq: With eosBeta, where do we call `producer.initTransactions` for the thread producer?

Reply:
Good catch -- missed that -- I think we can just make this call directly when we create the producer.